src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+        {
+            SqlInsertStructure sqlInsertStructure = new(entityName: entityName,
+                                                        sqlMetadataProvider: sqlMetadataProvider,
+                                                        authorizationResolver: _authorizationResolver,
+                                                        gQLFilterParser: _gQLFilterParser,
+                                                        mutationParams: parameters,
+                                                        httpContext: GetHttpContext(),
+                                                        isLinkingEntity: isLinkingEntity);
+
+            IQueryBuilder queryBuilder = _queryManagerFactory.GetQueryBuilder(sqlMetadataProvider.GetDatabaseType());
+            IQueryExecutor queryExecutor = _queryManagerFactory.GetQueryExecutor(sqlMetadataProvider.GetDatabaseType());
+
+            // When the entity is a linking entity, the higher level entity's name is used to get the
+            // datasource name. Otherwise, the entity's name is used.
+            string dataSourceName = isLinkingEntity ? _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(higherLevelEntityName)
+                                                    : _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(entityName);
+            string queryString = queryBuilder.Build(sqlInsertStructure);
+            Dictionary<string, DbConnectionParam> queryParameters = sqlInsertStructure.Parameters;
+
+            List<string> exposedColumnNames = new();
+            foreach (string columnName in sourceDefinition.Columns.Keys)
+            {
+                if (sqlMetadataProvider.TryGetExposedColumnName(entityName, columnName, out string? exposedColumnName) && !string.IsNullOrWhiteSpace(exposedColumnName))
+                {
+                    exposedColumnNames.Add(exposedColumnName);
+                }
+            }
+
+            DbResultSet? dbResultSet;
+            DbResultSetRow? dbResultSetRow;
+
+            dbResultSet = queryExecutor.ExecuteQuery(queryString,
+                                                     queryParameters,
+                                                     queryExecutor.ExtractResultSetFromDbDataReader,
+                                                     GetHttpContext(),
+                                                     exposedColumnNames,
+                                                     dataSourceName);
+
+            dbResultSetRow = dbResultSet is not null ? (dbResultSet.Rows.FirstOrDefault() ?? new DbResultSetRow()) : null;
+
+            if (dbResultSetRow is null || dbResultSetRow.Columns.Count == 0)
+            {
+                if (isLinkingEntity)
+                {
+                    throw new DataApiBuilderException(message: $"Could not insert row with given values in the linking table",
```

Comment:
Can we have the relationship name available to us? Currently, calling out just linking table seems not very helpful. Also, can we have a track of nesting levels wherever exceptions are thrown?